app/controllers/user.go

```diff
@@ -326,22 +338,68 @@ func (c Users) PassResetRequest() revel.Result {
 func (c Users) PassResetConfirm(userID uint, token, password string) revel.Result {
 	err := ds.PassResetConfirm(userID, token, password)
 	if err != nil {
-		return c.Error(err)
+		confirmed := false
+		return c.Render(err,confirmed)
+	}else {
+		confirmed := true
+		return c.Render(confirmed)
 	}
-	return c.OK()
+}
+
+func (c Users) PasswordReset(id uint, token string) revel.Result {
+	// req, err = http.NewRequest("POST", c.BaseUrl()mt+"/user/reset", c.RenderJson(reset))
+	// req.Header.Set("Content-Type", contentType)
+	// fmt.Println("EL pass: ", reset.Password)
+	return c.Render(id, token)
+}
+
+func (c Users) Confirm(id uint,token string) revel.Result {
+	// req, err = http.NewRequest("POST", c.BaseUrl()mt+"/user/reset", c.RenderJson(reset))
+	// req.Header.Set("Content-Type", contentType)
+	// fmt.Println("EL pass: ", reset.Password)
+	err := ds.ConfirmEmail(id, token)
+
+	if err != nil {
+		confirmed := false
+		return c.Render(err,confirmed)
+	}else {
+		confirmed := true
+		return c.Render(confirmed)
+	}
+}
+
+func ConfirmURL(userID uint, token string) (uri string) {
+	var host string
+	if revel.Server.Addr[0] == ':' {
+		host = "127.0.0.1" + revel.Server.Addr
+	} else {
+		host = revel.Server.Addr
+	}
+	host = revel.Config.StringDefault("server.host",host)
+	u := url.URL{}
+	u.Scheme = "http"
+	u.Host = host
+	u.Path = "/user/confirm"
+	q := u.Query()
+	q.Set("id", strconv.Itoa(int(userID)))
+	q.Set("token", token)
+	u.RawQuery = q.Encode()
+	return u.String()
 }
 
 func PasswordResetURL(userID uint, token string) (uri string) {
 	var host string
+	
 	if revel.Server.Addr[0] == ':' {
 		host = "127.0.0.1" + revel.Server.Addr
 	} else {
 		host = revel.Server.Addr
 	}
+	host = revel.Config.StringDefault("server.host",host)
```

Comment:
Doesn't seem to be working for me. The URLs contain the host port `45556` even though I have the API runing on port `8082`.